corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -282,13 +282,18 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
                 currentGroup = {
                     type: constants.GROUPED_QUESTION_TILE_ROW_TYPE,
                     children: [],
+                    ix: null,
```

Comment:
What is `ix`?

Reply:
For questions, group, and repeat groups, the ix indexes the order. So form in this order would have the associated ix
1. Question (ix = "0")
2. Question (ix = "1")
3. Group (ix ="2")
3.1 Question nested in group (ix="2,0")
3.2 Question nested in group (ix = "2,1")

The way i'm using it here isn't really following this convention, since this "Grouped question tile row type" is sort of imaginary. I defined ix to be the concatenation of ix contained in the group. So in the example, Questions 1 and 2 would be grouped, and create a "grouped question tile row type" with ix = "(0),(1)".

The purpose I added it is that for [knockout mapping plugin](https://knockoutjs.com/documentation/plugins-mapping.html), we use this ix was a "key" to identify objects to see if they need to be replaced or just updated

Reply:
That's pretty cool, I didn't realize we could/do use knockout mapping that way. I think your usage makes sense with the different types of nesting that already exist. Thanks for the explanation!